src/test/java/org/prebid/server/auction/PrivacyEnforcementServiceTest.java

```diff
@@ -0,0 +1,615 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anySet;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isNull;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+public class PrivacyEnforcementServiceTest extends VertxTest {
+
+    private final static String BIDDER_NAME = "someBidder";
+    private final static String BUYER_UID = "uidval";
+    private final static String PUBLISHER_ID = "pubId";
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private BidderCatalog bidderCatalog;
+    @Mock
+    private Metrics metrics;
+    @Mock
+    private GdprService gdprService;
+
+    private Clock clock;
+    private Timeout timeout;
+    private PrivacyEnforcementService privacyEnforcementService;
+
+    @Before
+    public void setUp() {
+        given(bidderCatalog.bidderInfoByName(anyString())).willReturn(givenBidderInfo(15, true));
+
+        clock = Clock.fixed(Instant.now(), ZoneId.systemDefault());
+        timeout = new TimeoutFactory(clock).create(500);
+
+        privacyEnforcementService = new PrivacyEnforcementService(gdprService, bidderCatalog, metrics, false);
+    }
+
+    @Test
+    public void shouldNotMaskWhenDeviceLmtIsNullAndExtRegsGdprIsOneAndGdprServiceRespondIsGdprEnforcedWithTrueAndResultByVendorWithoutEnforcement() {
+        // given
+        given(gdprService.isGdprEnforced(eq("1"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout)))
+                .willReturn(Future.succeededFuture(true));
+        given(gdprService.resultByVendor(eq(singleton(15)), eq("1"), any(), any(), eq(timeout)))
+                .willReturn(Future.succeededFuture(GdprResponse.of(true, singletonMap(15, true), null)));
+
+        final ExtUser extUser = ExtUser.builder().build();
+        final User user = notMaskedUser();
+        final Device device = notMaskedDevice();
+        final ExtRegs extRegs = ExtRegs.of(1);
+        final Regs regs = Regs.of(null, mapper.valueToTree(extRegs));
+        final Map<String, User> bidderToUser = singletonMap(BIDDER_NAME, user);
+
+        final BidRequest bidRequest = givenBidRequest(givenSingleImp(
+                singletonMap(BIDDER_NAME, 1)),
+                bidRequestBuilder -> bidRequestBuilder
+                        .user(user)
+                        .device(device)
+                        .regs(regs));
+
+        // when
+        final Future<Map<String, PrivacyEnforcementResult>> masked = privacyEnforcementService
+                .mask(bidderToUser, extUser, extRegs, singletonList(BIDDER_NAME), emptyMap(), bidRequest, PUBLISHER_ID, timeout);
+
+        // then
+        final PrivacyEnforcementResult expected = PrivacyEnforcementResult.of(user, device, regs);
```

Comment:
Please move variable to where it is - if verification fails it would never be used. Please check similar cases.